File: src/main/java/com/ccat/equipmentcalculator/model/GearSetResponse.java
package com.ccat.equipmentcalculator.model;

import com.ccat.equipmentcalculator.model.entity.CharacterClass;
import com.ccat.equipmentcalculator.model.entity.Item;

import java.util.List;

public class GearSetResponse {
    private Long id;

    private CharacterClass gearClass;

    private int itemLevel;
    private List<Item> equippedItems;

    //Constructors:
    public GearSetResponse() {

    }
    public GearSetResponse(Long id,CharacterClass gearClass, int itemLevel, List<Item> equippedItems) {
        this.id = id;
        this.gearClass = gearClass;
        this.itemLevel = itemLevel;
        this.equippedItems = equippedItems;
    }

    //Getters:
     public Long getId() {
        return id;
    }

    public CharacterClass getGearClass() {
        return gearClass;
    }

    public int getItemLevel() {
        return itemLevel;
    }

    public List<Item> getEquippedItems() {
        return equippedItems;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        GearSetResponse that = (GearSetResponse) o;

        if (itemLevel != that.itemLevel) return false;
        if (!id.equals(that.id)) return false;
        if (gearClass != that.gearClass) return false;
        return equippedItems.equals(that.equippedItems);
    }

    @Override
    public int hashCode() {
        int result = id.hashCode();
        result = 31 * result + gearClass.hashCode();
        result = 31 * result + itemLevel;
        result = 31 * result + equippedItems.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "GearSetResponse{" +
                "id=" + id +
                ", gearClass=" + gearClass +
                ", itemLevel=" + itemLevel +
                ", equippedItems=" + equippedItems +
                '}';
    }
}

File: src/main/java/com/ccat/equipmentcalculator/model/GearSetResponse.kt
package com.ccat.equipmentcalculator.model

import com.ccat.equipmentcalculator.model.entity.Item
import com.ccat.equipmentcalculator.model.entity.enums.CharacterClass

class GearSetResponse(
        val id: Long,

        val gearClass: CharacterClass,

        val itemLevel: Int,

        val equippedItems: List<Item>


) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as GearSetResponse

        if (id != other.id) return false
        if (gearClass != other.gearClass) return false
        if (itemLevel != other.itemLevel) return false
        if (equippedItems != other.equippedItems) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + gearClass.hashCode()
        result = 31 * result + itemLevel
        result = 31 * result + equippedItems.hashCode()
        return result
    }

    override fun toString(): String {
        return "GearSetResponse(id=$id, gearClass=$gearClass, itemLevel=$itemLevel, equippedItems=$equippedItems)"
    }

}